docs/user_guide/structure.md

```diff
@@ -0,0 +1,199 @@
+as analysis config.
+
+
+## General introduction
+
+Columnflow is a fully orchestrated columnar analysis tool for HEP analyses with pure Python. The
+workflow orchestration is managed by [law](https://github.com/riga/law) and
+[order](https://github.com/riga/order) is used for the metavariables. A short introduction to law is
+given in the {doc}`law section <law>`. If you have never used law before, this section is highly recommended as
+a few very convenient commands are presented there.
+
+
+The data processing in columnflow is based on columns in [awkward arrays](https://awkward-array.org/doc/main/)
+with [coffea](https://coffeateam.github.io/coffea/)-generated behaviour. Fields like "Jet" exist
+too, they contain columns with the same first dimension (the parameters of the field, e.g. Jet.pt).
+A few additional functions for simplified handling of columns were defined in
+{py:mod}`~columnflow.columnar_util`.
+
+As most of the information is conserved in the form of columns, it would be very inefficient
+(and might not even fit in the memory) to use all columns and all events from a dataset at once for
+each task. Therefore, in order to reduce the impact on the memory:
+- a chunking of the datasets is implemented using [dask](https://www.dask.org/): not all events
+from a dataset are inputed in a task at once, but only chunked in groups of events.
+(100 000 events max per group is default as of 05.2023, default is set in the law.cfg file).
+- the user needs to define for each {py:class}`~columnflow.production.Producer`,
+{py:class}`~columnflow.calibration.Calibrator` and {py:class}`~columnflow.selection.Selector` which
+columns are to be loaded (this happens by defining the ```uses``` set in the header of the
+decorator of the class) and which new columns/fields are to be saved in parquet files after the
+respective task (this happens by defining the ```produces``` set in the header of the decorator of
+the class). The exact implementation for this feature is further detailed in
+{doc}`building_blocks/selectors` and {doc}`building_blocks/producers`
+
+## Tasks in columnflow
+
+Tasks are [law](https://github.com/riga/law) objects allowing to control a workflow. All the tasks
+presented below are proposed by columnflow and allow for a fairly complete analysis workflow.
+However, as analyses are very diverse, it is possible that a specific analysis will need more
+stearing options or even completely new tasks. Thankfully, columnflow is not a fixed structure and
+you will be able to create new tasks in such a case, following the corresponding example in the
+{doc}`examples` section of this documentation.
+The full task tree of general columnflow tasks can be seen in
+[this wikipage](https://github.com/columnflow/columnflow/wiki#default-task-graph). Additionally, the
+CMS-specific {py:class}`~columnflow.tasks.cms.external.CreatePileUpWeights` task is implemented,
+but not present in the graph, as it is experiment specific.
```

Comment:
In my opinion, this task is really just there as part of the `pu_weight` producer, and if you don't use this producer, you also don't use the task. Of course this is also one interesting usecase (having a producer with custom requirements) we might want to show users somewhere, I just feel like it is a bit misplaced in the general introduction.
Do we plan on having a section on all the columnflow-provided producers? I feel like it would belong either there or as an example in some "special usecases" section

Reply:
done, for now there is a short description in the (newly created) cms-specializations section, which will be pushed with the next commits